pydicom/filereader.py

```diff
@@ -42,6 +42,59 @@
 sys_is_little_endian = (byteorder == 'little')
 
 
+def _auto_open(filepath, *args, **kwargs):
+    """Auto-magically (transparently) open a compressed file.
+
+    Supports `gzip` and `bzip2`.
+
+    Note: all compressed files should be opened as binary.
+    Opening in text mode is not supported.
+
+    Parameters
+    ----------
+    filepath : str
+        The file path.
+    *args : iterable
+        Positional arguments passed to `open()`.
+    **kwargs : dict
+        Keyword arguments passed to `open()`.
+
+    Returns
+    -------
+    file
+        A file object.
+
+    Raises
+    ------
+    IOError
+        on failure.
+
+    See Also
+    --------
+    open(), gzip.open(), bz2.open()
+    """
+    try:
```

Comment:
The real import happens only once. So there should not be a significant penalty hit if you happen to have `importlib`, `gzip` and `bz2`, which you should as they are standard modules.
In that scenario, I believe this is mostly a matter of style.
However, you are right in that some non-standard installation, and possibly even Py2.6 could have a performance it for always raising an import error (I am not sure that  `try` is substantially slower than a standard `if`, but this could well be).
If you prefer, I can move imports to the beginning of the file.